From 1a137c15a441ba1c3b0cbc402fafd21751be0ff9 Mon Sep 17 00:00:00 2001
From: Steven Moreland <smoreland@google.com>
Date: Fri, 9 May 2025 23:05:54 +0000
Subject: [PATCH] RPC Binder: shutdown on SPAN error.

This error return is not recoverable.

Bug: 416734088
Test: binderRpcTest
Change-Id: If0b8a8f36f797dcf927bfc2b5ae51e37e915f2f6
Merged-In: If0b8a8f36f797dcf927bfc2b5ae51e37e915f2f6
(cherry picked from commit dddbc115f88d99379ad9e118e551924c84f16e3a)
---
 libs/binder/RpcState.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/libs/binder/RpcState.cpp b/libs/binder/RpcState.cpp
index 03d974d186..adea2b0334 100644
--- a/libs/binder/RpcState.cpp
+++ b/libs/binder/RpcState.cpp
@@ -977,6 +977,7 @@ processTransactInternalTailCall:
                       " objectTableBytesSize=%zu. Terminating!",
                       transactionData.size(), sizeof(RpcWireTransaction),
                       transaction->parcelDataSize, objectTableBytes->size);
+                (void)session->shutdownAndWait(false);
                 return BAD_VALUE;
             }
             objectTableSpan = *maybeSpan;
-- 
2.50.1.351.gcf940e82a1-goog

